tests/sqlite/sqlite.rs

```diff
@@ -798,20 +802,22 @@ async fn test_multiple_set_progress_handler_calls_drop_old_handler() -> anyhow::
 #[sqlx_macros::test]
 async fn test_query_with_update_hook() -> anyhow::Result<()> {
     let mut conn = new::<Sqlite>().await?;
-
+    static CALLED: AtomicBool = AtomicBool::new(false);
     // Using this string as a canary to ensure the callback doesn't get called with the wrong data pointer.
     let state = format!("test");
     conn.lock_handle().await?.set_update_hook(move |result| {
         assert_eq!(state, "test");
         assert_eq!(result.operation, SqliteOperation::Insert);
         assert_eq!(result.database, "main");
         assert_eq!(result.table, "tweet");
-        assert_eq!(result.rowid, 2);
+        assert_eq!(result.rowid, 3);
+        CALLED.store(true, Ordering::Relaxed);
     });
 
     let _ = sqlx::query("INSERT INTO tweet ( id, text ) VALUES ( 3, 'Hello, World' )")
         .execute(&mut conn)
         .await?;
+    assert!(CALLED.load(Ordering::Relaxed));
```

Comment:
Without these asserts, it's possible that these tests would still pass if the callback functions were never invoked. I went ahead and added these checks to the existing tests as well as the new tests that I created.